crates/typst/src/layout/inline/mod.rs

```diff
@@ -1079,13 +1106,21 @@ fn line<'a>(
         // need the shaped empty string to make the line the appropriate
         // height. That is the case exactly if the string is empty and there
         // are no other items in the line.
-        if hyphen || start + shaped.text.len() > range.end || maybe_adjust_last_glyph {
+        if hyphen
+            || start + shaped.text.len() > range.end
+            || maybe_adjust_last_glyph
+            || (prepend_hyphen && before.is_empty())
+        {
             if hyphen || start < range.end || before.is_empty() {
                 let mut reshaped = shaped.reshape(engine, &p.spans, start..range.end);
                 if hyphen || shy {
                     reshaped.push_hyphen(engine, p.fallback);
                 }
 
+                if prepend_hyphen && before.is_empty() {
+                    reshaped.prepend_hyphen(engine, p.fallback);
+                }
```

Comment:
I'm not too fond of the duplication in doing it potentially in this branch (for the last item) and then again in the later branch (for the first item, if first != last). Maybe we could just do it at the end, similarly to `maybe_adjust_first_glyph`?

Reply:
I have created a `do_prepend_hyphen` variable. So I can reuse it on both places.

Reply:
What I meant was more to have the `prepend_hyphen` itself just a single time right before `if maybe_adjust_first_glyph {` with a construction similar to the `first.as_mut().or(last.as_mut())` used there. Do you think that'd be possible and make sense?

Reply:
I think I got it now.